cpp/tests/io/json/json_writer.cpp

```diff
@@ -611,4 +613,29 @@ TEST_P(JsonCompressedWriterTest, Unicode)
   run_test(out_options, expected);
 }
 
+TEST_P(JsonCompressedWriterTest, UnicodeUnescaped)
+{
+  auto [tbl_view, mt] = make_unicode_table();
+
+  std::vector<char> out_buffer;
+  auto destination = cudf::io::sink_info(&out_buffer);
+  auto out_options = cudf::io::json_writer_options_builder(destination, tbl_view)
+                       .include_nulls(true)
+                       .metadata(mt)
+                       .lines(true)
+                       .na_rep("null")
+                       .utf8_unescaped(true)
+                       .build();
+
+  cudf::io::write_json(out_options, cudf::test::get_default_stream());
+  std::string const output_string(out_buffer.data(), out_buffer.size());
+
+  EXPECT_TRUE(output_string.find("ராபிட்ஸ்") != std::string::npos);
+  EXPECT_TRUE(output_string.find("𐐷") != std::string::npos);
+  EXPECT_TRUE(output_string.find("𝞵") != std::string::npos);
+
+  EXPECT_TRUE(output_string.find("\\\"") != std::string::npos);
+  EXPECT_TRUE(output_string.find("\\n") != std::string::npos);
```

Comment:
Can we just test with an exact expected string?

Reply:
Done.